app/components/Search/SearchResults.tsx

```diff
@@ -16,130 +14,100 @@ type SearchResultItemProps = {
 };
 
 type SearchResultProps = {
-  query: string;
   searching: boolean;
   results: SearchResult[];
-  navigationLinks: NavigationLink[];
-  adminLinks: NavigationLink[];
   onCloseSearch: () => void;
   selectedIndex: number;
 };
 
-const SearchResultItem = ({
-  result,
-  isSelected,
-  onCloseSearch,
-}: SearchResultItemProps) => (
-  <Link to={result.link} onClick={onCloseSearch}>
-    <li className={cx(isSelected && styles.isSelected, styles.resultItem)}>
-      {isUserResult(result) && (
+const toResultIcon = (result: SearchResult) => {
+  switch (result.iconType) {
+    case 'profilePic':
+      return (
         <ProfilePicture
           size={28}
           user={result}
-          style={{
-            margin: '0px 12px 0px 0px',
-          }}
+          className={styles.searchResultItemIcon}
         />
-      )}
-      {!isUserResult(result) && result.icon && (
+      );
+    case 'image':
+      if (result.icon) {
+        return (
+          <Image
+            alt={`${result.title}'s logo`}
+            src={result.icon!}
+            style={{ width: '28px', display: 'block' }}
+            className={styles.searchResultItemIcon}
+          />
+        );
+      }
+    default:
+    case 'icon':
+      return (
         <Icon
-          name={result.icon}
+          name={result.icon ?? 'help'}
           size={28}
           className={styles.searchResultItemIcon}
         />
-      )}
-      <ul>
-        <li className={styles.resultTitle}>
-          <div className={styles.truncateTitle}>{result.title}</div>
-        </li>
-        <li className={styles.resultDetails}>
+      );
+  }
+};
+
+export const SearchResultItem = ({
+  result,
+  isSelected,
+  onCloseSearch,
+}: SearchResultItemProps) => (
+  <Link to={result.link} onClick={onCloseSearch}>
+    <li className={cx(isSelected && styles.isSelected, styles.resultItem)}>
+      {toResultIcon(result)}
```

Comment:
Just an FYI, the more "react way" to do this is to make it a component and pass the result as a prop (even though this is technically equivalent)

Reply:
Good point!